### DISCUSSION TITLE
Strugling with API filters

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#4606] [Q&A] [answered]
I need some help with creating API filters.

collection user_roles 
collection projects has a column executor which have a relation to the user auth.

A user can have multiple roles from (collection user_roles). If he has the user role, I want him to only be able to see his own projects.  If he has a role other than user, he can see everything from the projects collection.

I came up with the followed filter but if I have a role other than user, I don't see other projects here
````@request.auth.id != "" && (@request.auth.functions.name ?= "user" && user = @request.auth.id)````

What i am doing wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm confused from the description and the attempted API rule. Is the `user` in `user = @request.auth.id` field is the same as the `executor` in _"projects has a column executor"_?

> If he has a role other than user, he can see everything from the projects collection.

I'm not sure that I understand your collections structure but changing the `&&` to `||` in the parenthesis expression and negating the "functions.name" check should satisfy your requirement (_unless I got that wrong too_):

```js
@request.auth.id != "" &&
(
    // doesn't have "functions" with name "user"
    @request.auth.functions.name != "user" ||
    // OR if it has "functions" user then require the "user" field to match with the current auth id
    user = @request.auth.id
)
```



If the above doesn't help, please paste your collections configuration as json from the _Admin UI > Settings > Export collections_.

**Comment by starredev** [comment]
> I'm confused from the description and the attempted API rule. Is the `user` in `user = @request.auth.id` field is the same as the `executor` in _"projects has a column executor"_?
> 
> > If he has a role other than user, he can see everything from the projects collection.
> 
> I'm not sure that I understand your collections structure but changing the `&&` to `||` in the parenthesis expression and negating the "functions.name" check should satisfy your requirement (_unless I got that wrong too_):
> 
> ```js
> @request.auth.id != "" &&
> (
>     // doesn't have "functions" with name "user"
>     @request.auth.functions.name != "user" ||
>     // OR if it has "functions" user then require the "user" field to match with the current auth id
>     user = @request.auth.id
> )
> ```
> 
> If the above doesn't help, please paste your collections configuration as json from the _Admin UI > Settings > Export collections_.

Sorry for the confusion, it is the same so I changed user to executor in your example and it works :) Thank you so much
